wallet/wallet.c

```diff
@@ -340,6 +342,201 @@ bool wallet_shachain_load(struct wallet *wallet, u64 id,
 	return true;
 }
 
+static bool sqlite3_column_short_channel_id(sqlite3_stmt *stmt, int col,
+					    struct short_channel_id *dest)
+{
+	const char *source = sqlite3_column_blob(stmt, col);
+	size_t sourcelen = sqlite3_column_bytes(stmt, col);
+	return short_channel_id_from_str(source, sourcelen, dest);
+}
+
+static bool sqlite3_column_sig(sqlite3_stmt *stmt, int col, secp256k1_ecdsa_signature *sig)
+{
+	u8 buf[64];
+	if (!sqlite3_column_hexval(stmt, col, buf, sizeof(buf)))
+		return false;
+	return secp256k1_ecdsa_signature_parse_compact(secp256k1_ctx, sig, buf) == 1;
+}
+
+static bool sqlite3_column_pubkey(sqlite3_stmt *stmt, int col,  struct pubkey *dest)
+{
+	u8 buf[PUBKEY_DER_LEN];
+	if (!sqlite3_column_hexval(stmt, col, buf, sizeof(buf)))
+		return false;
+	pubkey_from_der(buf, sizeof(buf), dest);
+	return true;
+}
+
+static u8 *sqlite3_column_varhexblob(tal_t *ctx, sqlite3_stmt *stmt, int col)
+{
+	const u8 *source = sqlite3_column_blob(stmt, col);
+	size_t sourcelen = sqlite3_column_bytes(stmt, col);
+	return tal_hexdata(ctx, source, sourcelen);
+}
+
+static bool wallet_peer_load(struct wallet *w, const u64 id, struct peer *peer)
+{
+	bool ok = true;
+	sqlite3_stmt *stmt = db_query(__func__, w->db, "SELECT id, node_id FROM peers WHERE id=%"PRIu64";", id);
+	if (!stmt || sqlite3_step(stmt) != SQLITE_ROW) {
+		sqlite3_finalize(stmt);
+		return false;
+	}
+	peer->unique_id = sqlite3_column_int64(stmt, 0);
+	ok &= sqlite3_column_pubkey(stmt, 1, &peer->id);
+	sqlite3_finalize(stmt);
+	return ok;
+}
+
+/**
+ * wallet_stmt2channel - Helper to populate a wallet_channel from a sqlite3_stmt
+ *
+ * Returns true on success.
+ */
+static bool wallet_stmt2channel(struct wallet *w, sqlite3_stmt *stmt,
+				struct wallet_channel *chan)
+{
+	bool ok = true;
+	int col = 0;
+	struct channel_info *channel_info;
+	struct sha256_double temphash;
+	struct short_channel_id scid;
+
+	if (!chan->peer) {
+		chan->peer = talz(chan, struct peer);
+	}
```

Comment:
I really dislike this pattern of overwriting if it exists, or allocating if it doesn't, but I guess it's temporary.

Eventually, we'll want to just remember the peerid and the first htlc deadline, and load the rest from the database only when we hand it to the subdaemon.  That's for later, but just so we know where we're going...